packages/api/src/models/security/denied-permissions.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Represents a map of denied paths.
+ * - Each key is a path that is denied.
+ * - The value is always `true`, indicating the path is denied.
+ */
+export class DeniedPermissions {
```

Comment:
There is no value DeniedPermissions to be a class in the way how it's used currently and is kind of misleading. If the idea was to just define the type, then better make it simply an interface or a type (`type DeniedPermissions = Record<string, true>`). If the impact in the legacy-workbech wouldn't be big, then my oppinion is that it'd be better if we back this by a map, and expose some methods for getting and setting values in the map inside, like `denyPermission(location: string): void` and `hasPermission(location: string): boolean`. This would encapsulate the logic and make it more flexible and typesafe.